subsys/debug/Kconfig

```diff
@@ -281,6 +284,34 @@ config OPENOCD_SUPPORT
 	  OpenOCD to determine the state of running threads.  (This option
 	  selects CONFIG_THREAD_MONITOR, so all of its caveats are implied.)
 
+config GDBSTUB
+	bool "GDB remote serial protocol support [EXPERIMENTAL]"
+	depends on ARCH_HAS_GDBSTUB
+	help
+	  This option enable support the target using GDB, or any other
+	  application that supports GDB protocol.
+
+if GDBSTUB
+
+choice
+	prompt "GDB backend"
+
+config GDBSTUB_SERIAL_BACKEND
+	bool "Use serial backend"
+	depends on SERIAL
+	help
+	  Use serial as backenf for GDB
+
+config GDBSTUB_SERIAL_BACKEND_NAME
```

Comment:
shouldn't this be outside of the choice?

Reply:
does it make sense ? I mean why allows set gdbstub serial backend name if serial backend is not selected ?

Reply:
Every `config` inside `choice`-`endchoice` pair becomes a select-able option, which means there will be two options to choose in menuconfig. So this one really needs to be moved outside of choice block.